harness/tests/experiment/fixtures/pytorch_onevar_model.py

```diff
@@ -237,14 +237,15 @@ class AMPTestDataset(OnesDataset):
         "large": 2e4,
     }
 
-    def __init__(self, stages: Iterable[str]) -> None:
+    def __init__(self, stages: Iterable[str], aggregation_freq: int = 1) -> None:
         self.stages = stages
+        self._agg_freq = aggregation_freq
 
     def __len__(self) -> int:
-        return len(self.stages)
+        return len(self.stages) * self._agg_freq
 
     def __getitem__(self, index: int) -> Tuple:
-        x = self.STAGE_DATUM[self.stages[index]]
+        x = self.STAGE_DATUM[self.stages[index // self._agg_freq]]
         return torch.Tensor([float(x)]), torch.Tensor([float(x)])
```

Comment:
What is the scope of this particular PR.  Do you aim to test distributed training too?  Or is that future work?

If you are aiming to test distributed training, I would challenge you to set up the dataset in a way that only one worker is hitting overflow events.

The failure case that concerns us is basically, if we mess up the gradient aggregation or the ordering between communicating gradients and stepping the optimizer, some scalers on some workers might overflow while other scalers on other workers might step normally.

So to write a test that ensures we don't do that, you will need a dataset where different workers see different data.

When all workers see identical data, the risk of a false positive test result is much higher.  But it definitely is a much harder test to write.

If you are ok leaving the distributed training work for another pr, I would be ok with that, and what you have here seems just fine to me.

Reply:
I can envision such a distributed test, but it will take some work and time for me to get close enough to grokking how Determined does distributed training.
So let's keep a smaller scope for this particular PR, and let's work on creating a new task for a distributed version of this test (I think a call is in order).